superset/databases/dao.py

```diff
@@ -33,6 +34,25 @@ class DatabaseDAO(BaseDAO):
     model_cls = Database
     base_filter = DatabaseFilter
 
+    @classmethod
+    def update(
+        cls,
+        model: Database,
+        properties: Dict[str, Any],
+        commit: bool = True,
+    ) -> Database:
+        if "encrypted_extra" in properties:
+            new = json.loads(properties["encrypted_extra"])
```

Comment:
shouldn't we have try/except here just incase we have malformed coming in from the modal?

Reply:
Yeah, good point, that's safer. Let me add.